storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -351,7 +359,12 @@ private void fixDeltaLog(FileSystem fs, ExternalCommitEntry entry) throws IOExce
                 fixDeltaLogPutCompleteDbEntry(entry);
                 LOG.info("fixed {}", entry.fileName);
                 return;
-            } catch(Throwable e) {
+            } catch (java.nio.file.FileAlreadyExistsException e) {
```

Comment:
`writeCopyTempFile` calls `copyFile`, which has the same issue (a concurrent reader or writer copies the file before `writeCopyTempFile` is finished). Should we just change `copyFile` instead?

Reply:
> copyFile does not throw FIleAlreadyExistsException?

This one

> copyFile copies with overwrite=true?

Not this one because we want to avoid overwrite if possible